packages/constellation/src/components/Compositions/Logo/Logo.tsx

```diff
@@ -0,0 +1,97 @@
+import * as Avatar from '@radix-ui/react-avatar'
+import cx from 'classnames'
+import React from 'react'
+import { Textfit } from 'react-textfit'
+
+import { InternalLogoComponent, LogoComponent } from './Logo.types'
+
+const InternalLogo: InternalLogoComponent = ({
+  circle,
+  className,
+  color,
+  icon,
+  imgAlt,
+  imgSrc,
+  text,
+}) => {
+  return (
+    <Avatar.Root
+      className={cx(
+        { '!rounded-full': circle },
+        'w-8 h-8 rounded flex justify-center items-center bg-primary relative overflow-hidden',
+        className,
+      )}
+      style={color && { backgroundColor: color }}
+    >
+      {text && (
+        <div className='w-[80%]'>
+          <Textfit mode='single' className='text-white-000'>
+            {text}
+          </Textfit>
+        </div>
+      )}
+      {icon}
+      {imgSrc && (
+        <>
+          <Avatar.Image src={imgSrc} className='w-full h-full object-cover' />
+          <Avatar.Fallback>
+            <div className='w-[80%]'>
+              <Textfit mode='single' className='text-white-000'>
+                {imgAlt}
+              </Textfit>
+            </div>
+          </Avatar.Fallback>
+        </>
+      )}
+    </Avatar.Root>
+  )
+}
+
+const Logo: LogoComponent = ({
+  backgroundColor,
+  circle = true,
+  primaryIcon,
+  primaryImgAlt,
+  primaryImgSrc,
+  primaryText,
+  secondaryIcon,
+  secondaryImgAlt,
+  secondaryImgSrc,
+  secondaryText,
+  secondaryVariant = 'primary',
```

Comment:
How would you feel about consolidating these props into `primaryIcon` and `secondaryIcon`, both of which could be objects typed something like:
```
interface LogoIcon {
  icon: ...
  imgAlt: string
  imgSrc: string
  imgText: string
  variant?:
}
```
I think it would make it slightly easier for the consumer to include/omit icons this way. What do you think?

Reply:
Great idea!